packages/graphql/tests/integration/math.int.test.ts

```diff
@@ -0,0 +1,809 @@
+                `,
+                {
+                    id,
+                    initialViewers,
+                    name
+                }
+            );
+            // Update movie
+             const gqlResult = await graphql({
+                schema: await neoSchema.getSchema(),
+                source: query,
+                variableValues: { id, value: 10 },
+                contextValue: { driver, driverConfig: { bookmarks: session.lastBookmark() } },
+            });
+
+            expect(gqlResult.errors).toBeUndefined(); 
+            const storedValue = await session.run(
+                `
+                MATCH (n:Actor {id: $id})--(m:Movie) RETURN n.name AS name, m.viewers AS viewers
+                `,
+                {
+                    id
+                }
+            );
+            expect(storedValue.records[0].get('viewers')).toEqual(int(110));
+            expect(storedValue.records[0].get('name')).toBe(name);
+
+        } finally {
+            await session.close();
+        }
+      });
+
+      test('Should throws an error if the property holds Nan values', async () => {
+        const session = driver.session();
+
+        const typeDefs = `
+        type Movie {
+            id: ID!
+            viewers: Int
+        }
+        `;
+
+        const neoSchema = new Neo4jGraphQL({ typeDefs });
+
+        const id = generate({
+            charset: "alphabetic",
+        });
+
+        const query = `
+        mutation($id: ID, $value: Int) {
+            updateMovies(where: { id: $id }, update: {viewers_INCREMENT: $value}) {
+                movies {
+                    id
+                    viewers
+                }
+            }
+        }
+        `;
+
+        try {
+            // Create new movie
+            await session.run(
+                `
+                CREATE (:Movie {id: $id})
+                `,
+                {
+                    id,
+                }
+            );
+            // Update movie
+            const gqlResult = await graphql({
+                schema: await neoSchema.getSchema(),
+                source: query,
+                variableValues: { id, value: 10 },
+                contextValue: { driver, driverConfig: { bookmarks: session.lastBookmark() } },
+            });
+
+            expect(gqlResult.errors).toBeDefined();
+            const storedValue = await session.run(
+                `
+                MATCH (n:Movie {id: $id}) RETURN n.viewers AS viewers
+                `,
+                {
+                    id
+                }
+            );
+            expect(storedValue.records[0].get('viewers')).toBeNull();
+
+        } finally {
+            await session.close();
+        }
+      });
+
+      test('Should be possible to update relationship properties', async () => {
+        const session = driver.session();
+        const initialPay = 100;
+        const payIncrement = 50;
+        const typeDefs = `
+        type Movie {
```

Comment:
We just discovered that in many test cases in this file we're using the `Movie` type, also as when creating the movie `CREATE (a:Movie {title: "The Matrix"}) ...`, which may lead to having several nodes in the database that are matching which can cause issues when running the int tests several times (locally).
To prevent this we use the `generateUniqueTypes` function, see an example here: https://github.com/neo4j/graphql/blob/dev/packages/graphql/tests/integration/issues/988.int.test.ts#L27
I know it's a bit of work to change it for all the occurrences in this file but it may well be worth it to prevent issues.

Reply:
Don't worry! I was already thinking to update this PR specifying the error type when it's possible as suggested by you!